sale_stock_picking_note/models/sale_stock.py

```diff
@@ -1,15 +1,26 @@
 # Copyright 2018 Tecnativa - Carlos Dauden
 # Copyright 2021 Tecnativa - Víctor Martínez
+# Copyright 2023 Moduon Team S.L.
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
-    picking_note = fields.Html(string="Picking Internal Note")
-    picking_customer_note = fields.Text(string="Picking Customer Comments")
+    picking_note = fields.Html(
+        string="Picking Internal Note",
+    )
+    picking_customer_note = fields.Text(
+        string="Picking Customer Comments",
+    )
+
+    @api.onchange("partner_id")
```

Comment:
suggestion: This will only work on UI. For a more resilient way use a storable not readonly compute.

Reply:
Changed to compute

Reply:
I'm not sure I prefer this new approach based on computes.

Imagine this flow:

1. Wood Corner has a note: "never deliver on sundays". 
2. They buy a new chair. However, this one is urgent, and it can be delivered on sunday. In the sales order, the salesman writes: "this time, deliver on sunday".
3. Then he realizes that this happens often, so he decides to go to the partner and change the note to: "unless explicitly stated, don't deliver on sundays".

If we use an onchange, all goes as expected. If we use a compute, then all open orders and pickings will get that last note, and it will override the value that the salesman already changed by hand.

Thus, I believe an onchange is a better option, as we only want to get some values *by default*, but still leave room for human intervention later.

Reply:
Sure, and that's why I commented not depending on the partner fields, so the compute behaves like an onchange.

In the other hand, I'm thinking in processes that create sale orders automatically (e.g.: purchase sale intercompany) that should seamlessly take those default notes. A compute is the way to go for this case, IMO

Reply:
Ah OK, with that solution, then it makes sense. Thanks!